pkgs/applications/editors/vim/plugins/utils/nvim-treesitter/update.py

```
@@ -1,66 +1,97 @@
+        if "url" not in install_info or "revision" not in install_info:
+            log.warning(f"Parser {lang} missing required URL or revision, skipping")
+            return ""
 
-    generated = f"""  {lang} = buildGrammar {{
+        url = install_info["url"]
+        rev = install_info["revision"]
+
+        generated = f"""  {lang} = buildGrammar {{
     language = "{lang}";
     version = "0.0.0+rev={rev[:7]}";
     src = """
 
-    generated += subprocess.check_output(["nurl", url, rev, "--indent=4"], text=True)
-    generated += ";"
+        generated += subprocess.check_output(["nurl", url, rev, "--indent=4"], text=True)
+        generated += ";"
 
-    location = info.get("location")
-    if location:
-        generated += f"""
+        # Handle location if present
+        location = install_info.get("location", "")
+        if location:
+            generated += f"""
     location = "{location}";"""
 
-    if info.get("requires_generate_from_grammar"):
-        generated += """
+        # Handle grammar generation flag
+        if install_info.get("generate", False):
+            generated += """
     generate = true;"""
 
-    generated += f"""
+        generated += f"""
     meta.homepage = "{url}";
   }};
 """
 
-    return generated
-
-
-def update_grammars(nvim_treesitter_dir: str):
-    """
-    The lockfile contains just revisions so we start neovim to dump the
-    grammar information in a better format
-    """
-    # the lockfile
-    cmd = [
-        "nvim",
-        "--headless",
-        "-u",
-        "NONE",
-        "--cmd",
-        f"set rtp^={nvim_treesitter_dir}",
-        "+lua io.write(vim.json.encode(require('nvim-treesitter.parsers').get_parser_configs()))",
-        "+quit!",
-    ]
-    log.debug("Running command: %s", ' '.join(cmd))
-    configs = json.loads(subprocess.check_output(cmd))
+        return generated
+    except Exception as e:
+        log.error(f"Error generating grammar for {lang}: {e}")
+        return ""
```

Let's not fail silently